src/OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule/TelemetryHttpModule.cs

```diff
@@ -100,7 +138,7 @@ private void Application_BeginRequest(object sender, EventArgs e)
         {
             var context = ((HttpApplication)sender).Context;
             AspNetTelemetryEventSource.Log.TraceCallback("Application_BeginRequest");
-            ActivityHelper.CreateRootActivity(context, this.ParseHeaders);
+            ActivityHelper.StartAspNetActivity(this.TextMapPropagator, context, this.OnRequestStartedCallback);
```

Comment:
not a blocking comment, just sharing some thoughts -- this somewhat makes this a "tracing" only module, as one cannot write a MetricInstrumentation (without tracing).

The CallBacks could be made independent of activity, so a metric-only user can leverage them and build metrics out without creating activity.

Reply:
Good call. I could drop the `Activity` param from the callbacks and we could get it via `Activity.Current` in AspNet instrumentation. What I was thinking wrt to metrics is we would add a Meter in here eventually and then some instruments which we would also listen to from our instrumentation. The goal being to have TelemetryHttpModule emitting traces and metrics for anyone that wants to listen. What do you think about that?

/cc @alanwest